### PR TITLE
Fix logrhythm integration

### PR DESCRIPTION
**Pull request by adi88d**
Fix logrhythm integration
Add context data to command

## Status
In progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/12093

### CODE REVIEW

**Comment by adi88d** [comment]
Test playbook updated

File: Integrations/integration-LogRhythm.yml [hunk 1]
```diff
@@ -545,9 +580,19 @@ script:
         var alarms = getAlarmsData(allUsers, count, pageSize, startDate, endDate, dataArgs.status, dataObj);
         if (alarms.length) {
             var md = tableToMarkdown('',  alarms);
-            return {Type: entryTypes.note, Contents: alarms, ContentsFormat: formats.json, HumanReadable: md};
+            return {
+                Type: entryTypes.note,
+                Contents: alarms,
+                ContentsFormat: formats.json,
+                HumanReadable: md,
+                EntryContext:{'Alarms' : alarms }
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]:
```suggestion
                EntryContext:{'LogRhythm.Alarm' : alarms }
```

File: Integrations/integration-LogRhythm.yml [hunk 2]
```diff
@@ -419,7 +423,13 @@ script:
         }
 
         var md = tableToMarkdown('LogRhythm added alarm comment ' + args.comment + ' for ID: ' + args['alarm-id'],  mdRaw);
-        return {Type: entryTypes.note, Contents: jsResponse, ContentsFormat: formats.json, HumanReadable: md};
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext:{'Alarm' : mdRaw }
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]:
```suggestion
            EntryContext:{'LogRhythm.Alarm' : mdRaw }
```

File: Integrations/integration-LogRhythm.yml [hunk 3]
```diff
@@ -439,7 +449,14 @@ script:
...
         }
         var md = tableToMarkdown('LogRhythm alarm for ID: ' + args['alarm-id'],  mdRaw);
-        return {Type: entryTypes.note, Contents: jsResponse, ContentsFormat: formats.json, HumanReadable: md};
+
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext:{'Alarm' : mdRaw }
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]:
```suggestion
            EntryContext:{'LogRhythm.Alarm' : mdRaw }
```

File: Integrations/integration-LogRhythm.yml [hunk 4]
```diff
@@ -461,7 +478,13 @@ script:
         }
         var processedMd = processAlarmEventsMd(mdRaw);
         var md = tableToMarkdown('LogRhythm alarm events for ID: ' + args['alarm-id'],  processedMd);
-        return {Type: entryTypes.note, Contents: jsResponse, ContentsFormat: formats.json, HumanReadable: md};
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext:{'AlarmEvent' : processedMd }
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]:
```suggestion
            EntryContext:{'LogRhythm.AlarmEvent' : processedMd }
```

File: Integrations/integration-LogRhythm.yml [hunk 5]
```diff
@@ -495,7 +518,13 @@ script:
             var mdRaw = dq(jsResponse, 'Envelope.Body.GetAlarmHistoryByIDResponse.GetAlarmHistoryByIDResult.Comments.AlarmCommentDataModel')
             md += tableToMarkdown('LogRhythm alarm comments history for ID: ' + args['alarm-id'],  mdRaw);
         }
-        return {Type: entryTypes.note, Contents: jsResponse, ContentsFormat: formats.json, HumanReadable: md};
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext:{'AlarmHistory' : mdRaw }
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]:
```suggestion
            EntryContext:{'LogRhythm.AlarmHistory' : mdRaw }
```

File: Integrations/integration-LogRhythm.yml [hunk 6]
```diff
@@ -517,7 +546,13 @@ script:
             delete mdRaw['-i'];
         }
         var md = tableToMarkdown('LogRhythm alarm status updated for alarm: *' + args['alarm-id']+'* status: *' + args.status +'*',  mdRaw);
-        return {Type: entryTypes.note, Contents: jsResponse, ContentsFormat: formats.json, HumanReadable: md};
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext:{'Alarm' : mdRaw }
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]:
```suggestion
            EntryContext:{'LogRhythm.Alarm' : mdRaw }
```

File: Integrations/integration-LogRhythm.yml [hunk 7]
```diff
@@ -539,15 +651,70 @@ script:
...
+            break;
+        }
+
+
+        if(timeFrame == 'Custom'){
+            startDate = new Date(dataArgs['start-date']).toISOString();
+            endDate = new Date(dataArgs['end-date']).toISOString();
+        }
+        else{
+            startDate = start.toISOString().slice(0,10);
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]: why do you need the slice here and not in the if-section?

File: Integrations/integration-LogRhythm.yml [hunk 8]
```diff
@@ -689,14 +922,17 @@ script:
...
+    - contextPath: LogRhythm.Alarm.Status
+      description: Alarm new status.
+    - contextPath: LogRhythm.Alarm.ID
+      description: The unique id of the alarm.
     description: Update alarm status
   - name: lr-get-alarms
     arguments:
     - name: start-date
-      required: true
       description: 'Start date for the data query. For example: start-date="2018-03-27"'
```
Comment (yaakovi) [inline Integrations/integration-LogRhythm.yml]: specify that this is ignore in case `time_frame` is not `Custom`

**Pull request merged**